### PR TITLE
fix: correctly parse oauth response

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

Parsing was broken

### CODE REVIEW

File: core/src/oauth/client.rs [hunk 1]
```diff
@@ -29,18 +35,18 @@ impl OauthClient {
...
-                    CredentialProvider::from_str(&json["provider"].as_str().ok_or_else(|| {
-                        anyhow!("Invalid response from `oauth`: missing provider")
-                    })?)?;
-
-                let content = serde_json::from_value::<serde_json::Map<String, serde_json::Value>>(
-                    json["content"].clone(),
-                )
-                .map_err(|e| anyhow!("Failed to parse content: {}", e))?;
+                let response = serde_json::from_str::<APIResponse>(&body)?
+                    .response
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply core/src/oauth/client.rs:39]: we already check status code 200

File: core/src/utils.rs [hunk 2]
```diff
@@ -112,13 +112,13 @@ pub fn confirm(msg: &str) -> Result<bool> {
...
-#[derive(Serialize)]
+#[derive(Serialize, Deserialize)]
 pub struct APIError {
     pub code: String,
     pub message: String,
 }
 
-#[derive(Serialize)]
+#[derive(Serialize, Deserialize)]
 pub struct APIResponse {
```
Comment (Fraggle) [inline core/src/utils.rs:122]: I didn't know we already had that, i introduced something similar in my integration tests.

**Review by Fraggle — APPROVED** [review]

File: core/src/oauth/client.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Fraggle) [reply core/src/oauth/client.rs:39]: I would say that nothing prevent someone to write something that returns 200 and an error at the same time (that would be bad, agreed). I can understand both sides on this. No strong opinion.

File: core/src/utils.rs (hunk 2 above)
Comment (Fraggle) [reply core/src/utils.rs:122]: yeah, it was stupid of me. Will refactor to use it (unless you want to do it in this PR)

Comment (fontanierh) [reply core/src/utils.rs:122]: Def not stupid, it wasn't designed for that. now that we have rust-to-rust it starts making sense

File: core/src/oauth/client.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (fontanierh) [reply core/src/oauth/client.rs:39]: I'll add the check, it's a better to know if that happens

**Pull request merged**